force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +132,16 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+        String cs1 = '';
+        String cs2 = '';
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        for (String i : scientistNames) {
+            if (i.containsIgnoreCase('g')) {
+              cs1 += i + ',';
+            }
+            cs2 = cs1.removeEnd(',');
+        }
+         return cs2; 
     }
```

Comment:
These variable names are pretty tough to understand. Always try to write code that the next developer can read easily. I accept it as an index variable in a for loop (an ancient CS convention), but that is about it. 

```
 public static String findTheScientist() {
        // The list of scientists' names
        List<String> scientistNames = new List<String>{
                'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'
        };
        String result = '';
        for (String name : scientistNames) {
            if (name.toLowerCase().contains('g')) {
                result += name + ',';
            }
        }
        return result.removeEnd(',');
    }
 ```